README.md

```diff
@@ -1,17 +1,107 @@
 # Exercises
 
-__NOTE:__ DO NOT USE THE BUILT IN METHODS WHEN ASKED TO MAKE THEM! 
+__NOTE:__ DO NOT USE THE BUILT IN METHODS WHEN ASKED TO MAKE THEM!
 
 1. Using Array.prototype add a new method `myForEach` that mimics the behaivor of the built in `forEach`.
 
-2. Use your `myForEach` to add `myMap` to the Array prototype. `myMap` should behave the same as regular `map`. 
+```js
+Array.prototype.myForEach = function(myFunc) {
+  for (let i = 0; i < this.length; i++) {
+    myFunc(this[i])
+  }
+}
+```
 
-3. Write a 'myFilter` that behave's the same as `filter`. 
+2. Use your `myForEach` to add `myMap` to the Array prototype. `myMap` should behave the same as regular `map`.
+```js
+Array.prototype.myForEach = function(myFunc) {
+  for (let i = 0; i < this.length; i++) {
+    myFunc(this[i])
+  }
+}
+Array.prototype.myMap = function(callback) { //callback - is the name of the function to be called
+let result = [];
+ this.myForEach(el => {
+  result.push(callback(el));
+ })
+return result;
+}
+const double = function(item) {
+  return item*2;
+}
+console.log([1,2,3].myMap(double))
+```
 
-4. Write a `myEvery`. 
+3. Write a `myFilter` that behaves the same as `filter`.
+```js
+//myFilter is testing a condition for Each of the elements in the array, and creates a new array with those who answered the condition (returned true). the condition can be stored as a function, and then be called back.
+const isOdd = function(num) {
+  return num%2!==0;
+}
+Array.prototype.myFilter = function(func) {
+let output = [];
+ for (let i = 0; i<this.length; i++) {
+   let result = func(this[i]);
+   if (result) {
+     output.push(this[i]);
+   }
+ }
+ return output;
+}
+console.log(isOdd(4))
+console.log([1,2,3,4].myFilter(isOdd))
+```
 
-5. Write a `myReduce`. It should use the first element in the array if none is given. 
+4. Write a `myEvery`.
+```js
+const isOdd = function(num) {
+  return num%2!==0;
+}
+Array.prototype.myEvery = function(func) {
```

Comment:
This is a neat approach. You can improve it by returning false at first false find. This will speed it up.